numpy/ma/core.py

```diff
@@ -5213,7 +5213,8 @@ def round(self, decimals=0, out=None):
             out.__setmask__(self._mask)
         return out
 
-    def argsort(self, axis=None, kind='quicksort', order=None, fill_value=None):
+    def argsort(self, axis=-1, kind='quicksort', order=None,
+                endwith=True, fill_value=None):
```

Comment:
Not changing it breaks liskov substitution though. The default values should really match `ndarray`. Fair call on swapping argument order